src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php

```diff
@@ -69,9 +69,19 @@ public function buildForm(FormBuilderInterface $builder, array $options)
 
                 $form = $event->getForm();
                 $formData = $form->getData();
+
                 if ($formData !== null) {
                     $formData->setType($data['type']);
                     $formData->setConfiguration($data['configuration']);
+
+                    if ($data['type'] === CatalogPromotionActionInterface::TYPE_FIXED_DISCOUNT) {
+                        foreach ($data['configuration'] as $channelConfiguration) {
+                            if ($channelConfiguration['amount'] === '') {
+                                return;
+                            }
+                        }
+                    }
```

Comment:
Should we be so hardcoded to one action type?

Reply:
thats only place where this error occurs, so i had to handle only when channelConfiguration is available.